src/pybind/mgr/dashboard/frontend/cypress/integration/orchestrator/workflow/nfs/nfs-export.e2e-spec.ts

```diff
@@ -0,0 +1,87 @@
+import { ServicesPageHelper } from 'cypress/integration/cluster/services.po';
+import { BucketsPageHelper } from 'cypress/integration/rgw/buckets.po';
+
+import { NFSPageHelper } from './nfs-export.po';
+
+describe('nfsExport page', () => {
+  const nfsExport = new NFSPageHelper();
+  const services = new ServicesPageHelper();
+  const buckets = new BucketsPageHelper();
+  const bucket_name = 'e2ebucket';
+  const fsPseudo = '/fspseudo';
+  const rgwPseudo = '/rgwpseudo';
+  const editpseudo = '/editpseudo';
+  const backends = ['CephFS', 'Object Gateway'];
+  const squash = 'no_root_squash';
+  const client: object = { addresses: '192.168.0.10' };
+
+  beforeEach(() => {
+    cy.login();
+    Cypress.Cookies.preserveOnce('token');
+    nfsExport.navigateTo();
+  });
+
+  describe('breadcrumb test', () => {
+    it('should open and show breadcrumb', () => {
+      nfsExport.expectBreadcrumbText('NFS');
+    });
+  });
+
+  describe('Create, edit and delete', () => {
+    it('should create an NFS cluster', () => {
+      services.navigateTo('create');
+
+      services.addService('nfs');
+      cy.wait(15000);
+    });
+
+    it('should create a nfs-export with CephFS backend', () => {
+      nfsExport.navigateTo();
+      nfsExport.existTableCell(fsPseudo, false);
+      nfsExport.navigateTo('create');
+      nfsExport.create(backends[0], squash, client, fsPseudo);
+      nfsExport.existTableCell(fsPseudo);
+    });
+
+    it('should show Clients', () => {
+      nfsExport.clickTab('cd-nfs-details', fsPseudo, 'Clients (1)');
+      cy.get('cd-nfs-details').within(() => {
+        nfsExport.getTableCount('total').should('be.gte', 0);
+      });
+    });
+
+    it('should edit an export', () => {
+      nfsExport.editExport(fsPseudo, editpseudo);
+
+      nfsExport.existTableCell(editpseudo);
+    });
+
+    it('should create a nfs-export with RGW backend', () => {
+      services.navigateTo('create');
+
+      services.addService('rgw');
+      cy.wait(20000);
+
+      buckets.navigateTo('create');
+      buckets.create(bucket_name, 'dashboard', 'default-placement');
+
+      nfsExport.navigateTo();
+      nfsExport.existTableCell(rgwPseudo, false);
+      nfsExport.navigateTo('create');
+      nfsExport.create(backends[1], squash, client, rgwPseudo, bucket_name);
+      nfsExport.existTableCell(rgwPseudo);
+    });
+
+    it('should delete exports, bucket and services', () => {
+      nfsExport.delete(editpseudo);
+      nfsExport.delete(rgwPseudo);
+
+      buckets.navigateTo();
+      buckets.delete(bucket_name);
+
+      services.navigateTo();
+      services.deleteService('rgw.foo', 5000);
+      services.deleteService('nfs.testnfs', 5000);
+    });
+  });
+});
```

Comment:
these specs are not yet running in the jenkins dashboard cephadm tests right? are we waiting for the cluster-expansion workflow to go in to trigger these jobs?

Reply:
This e2e needs --expanded option to setup the whole cluster, but currently jenkins job doesn't use it yet. So for now I'm locally verifying, not sure if there's a plan to include --expanded in jenkins job as well.. @alfonsomthd ?

Reply:
I don't think its a good idea to use the --expanded option on jenkins job. Its better to use the UI to setup whatever it is required for the NFS to start. Similar to what we've been doing on the cluster expansion. So I would suggest to run this test after the cluster expansion workflow test. That way we'll have a working cluster with us and then it would be just creating a filesystem and I believe we can do it simply from the UI itself.

Reply:
@avanthakkar @nizamial09 ideally Cluster expansion is to be merged before this, so after E2E wizard tests are done, then these ones can run: create an NFS service, then RGW service and finally creating an export.
@nizamial09 Bear in mind that right no we cannot create a filesystem from UI.

Reply:
> Bear in mind that right no we cannot create a filesystem from UI.

I wrongly assumed we can. Hmm, in that case we need to rely on the script to create a file system for us I think.